lxd/devices.go

```diff
@@ -484,6 +484,12 @@ func deviceTaskBalance(s *state.State) {
 			}
 		}
 
+		cpuPinStrategy := conf["limits.cpu.pin_strategy"]
+		cpuPinningSpecified := strings.ContainsAny(cpulimit, ",-")
+		if c.Type() == instancetype.VM && (cpuPinStrategy == "none" || cpuPinStrategy == "") && !cpuPinningSpecified {
```

Comment:
Would this be more concise?

```go
if c.Type() == instancetype.VM && cpuPinStrategy != "auto" && !cpuPinningSpecified {
```

Reply:
I like your suggestion for conciseness, but it's less descriptive. I'll add a comment for clarity as well - best of both approaches!